bpf/conntrack/cleanup.go

```diff
@@ -318,18 +322,29 @@ func NewStaleNATScanner(frontendHasBackend NATChecker) EntryScanner {
 			// Because client IP/Port are both in fwd key and rev key, we can
 			// can tell which one it is and thus determine exactly meaning of
 			// the other values.
-			if kA.Equal(revA) {
+			if kA.Equal(revA) && kAport == revAport {
 				epIP = revB
-				epPort = revKey.PortB()
+				epPort = revBport
 				svcIP = kB
-				svcPort = k.PortB()
-			} else if kA.Equal(revB) {
+				svcPort = kBport
+			} else if kB.Equal(revA) && kBport == revAport {
+				epIP = revB
+				epPort = revAport
+				svcIP = kA
+				svcPort = kAport
+			} else if kA.Equal(revB) && kAport == revBport {
 				epIP = revA
-				epPort = revKey.PortA()
+				epPort = revAport
 				svcIP = kA
-				svcPort = k.PortA()
+				svcPort = kAport
+			} else if kB.Equal(revB) && kBport == revBport {
+				epIP = revA
+				epPort = revBport
+				svcIP = kB
```

Comment:
Think this should be `kA` since `kB.Equal(revB)` means that `kB` is the source pod IP.